### PR TITLE
test: add DB-failure-path and edge-case coverage to OPDS catalog (#1774)

### PR DESCRIPTION
**Pull request by roberte777**
## Summary
- Closes #1774
- Adds the missing DB-failure-path (`internal(...)`/5xx) coverage for the five OPDS handlers named in the issue's evidence: `new::new_arrivals`, `search::search`, `authors::letter_index`, `authors::by_letter`, `authors::acquisition_feed`. Each closes the pool and drives the handler directly with a hand-built `OpdsAuthUser`, mirroring the existing `admin_sessions/tests.rs` pattern — closing the pool before a routed request would fail the auth extractor's own session lookup rather than the handler body under test.
- Adds `authors_by_letter_rejects_a_single_non_alphabetic_non_hash_character` covering `by_letter`'s second 400 branch (e.g. `"1"`, `"!"`).
- Adds two tests for `entry_updated`'s documented fallback to the current instant on a missing or malformed `added_at`.
- Test-only; no production code changes.

## Test plan
- [x] `cargo fmt` — clean
- [x] `cargo clippy -p omnibus --all-targets -- -D warnings` — clean
- [x] `cargo test -p omnibus opds` — 55 passed (including the 8 new tests), 0 failed
- [x] `cargo test -p omnibus` — 791 passed, 2 failed. The 2 failures (`backend::uploads::tests::commit_rejects_read_only_library_with_400`, `audiobook_commit_rejects_read_only_library_with_400`) are pre-existing and unrelated to this change: they chmod a directory to `0o555` and expect a `PermissionDenied` write failure, but this sandbox runs the test suite as `root`, which bypasses filesystem permission bits (root got `201 Created` instead of the expected `400`). Neither test is touched by this PR's diff.

## Version
- [x] **No release** — test-only change, no user-facing behavior change.

## Notes
- Suggested approach from the issue (hand-built extractor + closed pool, following `admin_sessions/tests.rs`) matched exactly since `OpdsAuthUser`'s own session lookup shares the `sessions` table with several of the handlers' first DB call, the same problem `admin_sessions/tests.rs`'s `fake_admin` helper already solves.

---
_Generated by [Claude Code](https://claude.ai/code/session_01DQAjLMLTEJqehG7NmQA7Z9)_

### CODE REVIEW

File: server/src/backend/opds/tests.rs [hunk 1]
```diff
@@ -348,6 +431,45 @@ async fn author_acquisition_feed_returns_404_for_an_unknown_author_id() {
...
+    let before = time::OffsetDateTime::now_utc();
+    let result = entry_updated(None);
+    let after = time::OffsetDateTime::now_utc();
+    let parsed =
+        time::OffsetDateTime::parse(&result, &time::format_description::well_known::Rfc3339)
+            .unwrap();
+    assert!(
+        parsed >= before && parsed <= after,
+        "expected {parsed} between {before} and {after}"
+    );
```
[the remark this answers is not in the document]
Comment (roberte777) [reply server/src/backend/opds/tests.rs:456]: Good catch — fixed in a9c5614f. Both `entry_updated_falls_back_to_the_current_instant_for_a_missing_timestamp` and `_for_a_malformed_timestamp` now bracket with a 2-second tolerance on each side (`before - Duration::seconds(2)` / `after + Duration::seconds(2)`) instead of the exact `now_utc()` bounds, so a small NTP adjustment or scheduling jitter between the two clock reads won't flake the assertion, while still catching a result that's off by more than that. Confirmed `cargo fmt`, `cargo clippy -p omnibus --all-targets -- -D warnings`, and `cargo test -p omnibus opds` (55/55) all pass with the change.

---
_Generated by [Claude Code](https://claude.ai/code)_
